### DISCUSSION TITLE
Gen 1 Hue Dimmer Button Mapping Correction in Homebridge/Homekit

### DISCUSSION THREAD

**Discussion by perfecblue** [Koenkk/zigbee2mqtt#7262] [Devices]
Hi, I'm trying to use the Gen 1 Hue Dimmer (Philips 324131092621) with a CC2652p-based Zigbee Gateway on Homebridge and trying to get this remote into Homekit.  The Dimmer successfully paired and with the "legacy: true" default was able to be seen and operated in Homekit.  However, I realized quickly that the actually button mapping of the Dimmer when recognized by the HomebridgeZ2M plugin was as follows:

Button 1 (down):
        - SINGLE: down-press
        - LONG  : down-hold
Button 2 (off):
        - SINGLE: off-press
        - LONG  : off-hold
Button 3 (on):
        - SINGLE: on-press
        - LONG  : on-hold
Button 4 (up):
        - SINGLE: up-press
        - LONG  : up-hold

This seems odd to me, since the actual button placement on the remote from top to bottom is ON - UP - DOWN - OFF.  I understand the support of this device was added a long time ago, but is there anyway I can change the button mapping when it was brought into Homebridge?

Thank you very much!

**Comment by perfecblue** [comment]
I just realized this is probably a HomebridgeZ2M issue, I've also reported it over there.  Still appreciate any helpful input on this issue here, thanks!

**Comment by shamimmirzai** [comment]
@perfecblue  I realise I'm coming in to this a few years later, but did you get any solution to this?
